syscall_sysv.go

```diff
@@ -47,10 +88,22 @@ func NewCallback(fn interface{}) uintptr {
 // only increase this if you have added more to the callbackasm function
 const maxCB = 2000
 
-var cbs struct {
-	lock  sync.Mutex
-	numFn int                  // the number of functions currently in cbs.funcs
-	funcs [maxCB]reflect.Value // the saved callbacks
+var cbs = struct {
+	lock     sync.RWMutex
+	holes    map[int]struct{}
+	funcs    [maxCB]reflect.Value // the saved callbacks
+	knownFn  map[uintptr]uintptr
+	knownIdx map[uintptr]int
```

Comment:
Could you add some comments what these do? knownFn and knownIdx are a bit vague on first glance